src/io/RootImporter.cc

```diff
@@ -85,55 +99,56 @@ std::shared_ptr<ParticleParams> RootImporter::load_particle_data()
-        particle_def.decay_constant
-            = (particle.is_stable ? ParticleDef::stable_decay_constant()
-                                  : 1. / particle.lifetime);
-        defs[i].second = std::move(particle_def);
+        defs[i].mass           = units::MevMass{particle.mass};
+        defs[i].charge         = units::ElementaryCharge{particle.charge};
+        defs[i].decay_constant = (particle.is_stable
+                                      ? ParticleDef::stable_decay_constant()
+                                      : 1. / particle.lifetime);
     }
 
-    // Sort by increasing mass, then by PDG code. Placing lighter particles
+    // Sort by increasing mass, then by PDG code (positive before negative of
+    // the same absolute value). Placing lighter particles
     // (more likely to be created by various processes, so more "light
     // particle" tracks) together at the beginning of the list will make it
     // easier to human-read the particles while debugging, and having them
     // at adjacent memory locations could improve cacheing.
-    std::sort(defs.begin(), defs.end(), [](const auto& lhs, const auto& rhs) {
-        return std::make_tuple(lhs.second.mass, lhs.first.pdg_code)
-               < std::make_tuple(rhs.second.mass, rhs.first.pdg_code);
-    });
+    auto to_particle_key = [](const auto& inp) {
+        int pdg = inp.pdg_code.get();
+        return std::make_tuple(inp.mass, std::abs(pdg), pdg < 0);
+    };
+    std::sort(defs.begin(),
+              defs.end(),
+              [to_particle_key](const auto& lhs, const auto& rhs) {
+                  return to_particle_key(lhs) < to_particle_key(rhs);
+              });
 
     // Construct ParticleParams from the definitions
     return std::make_shared<ParticleParams>(std::move(defs));
 }
 
 //---------------------------------------------------------------------------//
 /*!
- * Load all ImportPhysicsTable objects from the ROOT file as a vector
+ * Load all ImportProcess objects from the ROOT file as a vector
  */
-std::shared_ptr<std::vector<ImportPhysicsTable>>
-RootImporter::load_physics_table_data()
+std::vector<ImportProcess> RootImporter::load_processes()
 {
-    CELER_LOG(status) << "Loading physics data";
+    CELER_LOG(status) << "Loading physics processes";
     // Open tables branch
-    std::unique_ptr<TTree> tree_tables(root_input_->Get<TTree>("tables"));
+    std::unique_ptr<TTree> tree_tables(root_input_->Get<TTree>("processes"));
     CHECK(tree_tables);
     CHECK(tree_tables->GetEntries());
 
     // Load branch
-    ImportPhysicsTable  a_table;
-    ImportPhysicsTable* temp_table_ptr = &a_table;
-    tree_tables->SetBranchAddress("ImportPhysicsTable", &temp_table_ptr);
+    ImportProcess  a_table;
+    ImportProcess* temp_table_ptr = &a_table;
+    tree_tables->SetBranchAddress("ImportProcess", &temp_table_ptr);
 
-    std::vector<ImportPhysicsTable> tables;
+    std::vector<ImportProcess> tables;
```

Comment:
Maybe a nitpick, but now that the class has changed, we might update the variables names for readability (maybe `a_process` / `temp_process_ptr` / `processes`)

Reply:
Not a nitpick, a good catch.